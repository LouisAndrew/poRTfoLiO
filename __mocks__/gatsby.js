const React = require('react')
const gatsby = jest.requireActual('gatsby')

const mockSocials = [
    { link: 'http://localhost', social_key: 'github' },
    { link: 'http://google.com', social_key: 'dribbble' },
]

const mockButtonText = 'Click me'
const mockContactContent = 'Content'
const mockContactHeadline = 'Headline'
const mockEmail = 'john@doe.com'

module.exports = {
    ...gatsby,
    graphql: jest.fn(),
    Link: jest.fn().mockImplementation(
        // these props are invalid for an `a` tag
        ({
            activeClassName,
            activeStyle,
            getProps,
            innerRef,
            ref,
            replace,
            to,
            ...rest
        }) =>
            React.createElement('a', {
                ...rest,
                href: to,
            })
    ),
    StaticQuery: jest.fn(),
    // can I mock return value here?
    useStaticQuery: jest.fn(() => ({
        socialData: {
            frontmatter: {
                social: mockSocials,
            },
        },
        emailData: {
            frontmatter: {
                contact: {
                    button_text: mockButtonText,
                    contact_content: mockContactContent,
                    contact_headline: mockContactHeadline,
                    email: mockEmail,
                },
            },
        },

        // forced to write the hero-page-query mock here.
        heroData: {
            frontmatter: {
                hero: {
                    hero_content: 'Hello, World!', // hard coded from hero.test.tsx
                    hero_headline: 'I am headline', // Also hard coded from hero.test.tsx
                },
            },
        },
        imgS: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgM: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgL: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgXL: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
    })),
}
