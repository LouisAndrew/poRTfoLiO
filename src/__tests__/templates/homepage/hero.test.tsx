import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Hero from 'templates/homepage/hero'
// import '/__mocks__/window.js'

import * as Gatsby from 'gatsby'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

const mockHeroContent = 'Hello, World!'
const mockHeroHeadline = 'I am headline'

describe('Hero', () => {
    const Element = <Hero />

    afterEach(cleanup)
    beforeEach(() => {
        useStaticQuery.mockImplementationOnce(() => ({
            heroData: {
                frontmatter: {
                    hero: {
                        hero_content: mockHeroContent,
                        hero_headline: mockHeroHeadline,
                    },
                },
            },
            imgS: {
                frontmatter: {
                    hero: {
                        hero_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
            imgM: {
                frontmatter: {
                    hero: {
                        hero_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
            imgL: {
                frontmatter: {
                    hero: {
                        hero_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
            imgXL: {
                frontmatter: {
                    hero: {
                        hero_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
        }))

        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(), // deprecated
                removeListener: jest.fn(), // deprecated
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            })),
        })
    })

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(Element, div)
    })

    it('Should render the headline and content correctly', () => {
        const { queryByText, getByTestId } = render(Element)

        const headline = queryByText(mockHeroHeadline)
        const content = getByTestId('content')

        expect(headline).toBeInTheDocument()
        expect(content).toHaveTextContent(mockHeroContent)
    })

    it('matches snapshot', () => {
        const run = true

        if (run) {
            const tree = renderer.create(Element).toJSON()
            expect(tree).toMatchSnapshot()
        }
    })
})
