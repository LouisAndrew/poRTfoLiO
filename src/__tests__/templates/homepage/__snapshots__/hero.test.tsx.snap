// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero matches snapshot 1`] = `
<div
  className="css-4cffwv"
  id="hero"
>
  <div
    className="css-104ytwb"
  >
    <div
      className="hero__flex-wrapper css-1d2tbd8"
    >
      <div
        className="css-1rikcx"
      >
        <div
          className=" gatsby-image-wrapper"
          style={
            Object {
              "display": "inline-block",
              "height": undefined,
              "overflow": "hidden",
              "position": "relative",
              "width": undefined,
            }
          }
        >
          <picture>
            <source
              media="(max-width: 640px)"
            />
            <source
              media="(max-width: 840px) and (max-height: 420px) and (orientation: landscape)"
            />
            <source
              media="(min-width: 1952px)"
            />
            <source />
            <img
              alt="Cool-looking emoji of me"
              loading="lazy"
              onLoad={[Function]}
              style={
                Object {
                  "height": "100%",
                  "left": 0,
                  "objectFit": "cover",
                  "objectPosition": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transition": "opacity 500ms",
                  "width": "100%",
                }
              }
            />
          </picture>
          <noscript
            dangerouslySetInnerHTML={
              Object {
                "__html": "<picture><source media=\\"(max-width: 640px)\\" srcset=\\"undefined\\" /><source media=\\"(max-width: 840px) and (max-height: 420px) and (orientation: landscape)\\" srcset=\\"undefined\\" /><source media=\\"(min-width: 1952px)\\" srcset=\\"undefined\\" /><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"Cool-looking emoji of me\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
              }
            }
          />
        </div>
      </div>
      <div
        className="hero__details-box css-dg0yw"
      >
        <h1
          className="css-b5c2cb"
          data-testid="heading"
        >
          Hello
           
          <span
            id="hello"
            role="img"
            style={
              Object {
                "display": "inline-block",
                "transition": "200ms",
              }
            }
          >
            ðŸ‘‹
          </span>
          <br />
          I am headline
          <br />
        </h1>
        <p
          className="css-vurnku"
          data-testid="content"
        >
          Hello, World!
        </p>
      </div>
    </div>
    <div
      className="scroller css-1sfhpft"
      role="complementary"
    >
      <h4
        className="css-vurnku"
      >
        SCROLL DOWN TO SEE MY PROJECTS
      </h4>
      <div
        className="scroller__box css-1o8ebip"
        role="button"
      >
        <div
          className="scroller__thumb css-atwtvm"
          data-testid="scroller-thumb"
        />
      </div>
    </div>
  </div>
</div>
`;
